### PR TITLE
enable seeding of GLOBAL_RNG when running tests

### PR DESCRIPTION
**Pull request by rfourquet**
When a test fails, the seed `failseed` of the global RNG is printed. It's then possible to reproduce the failure via `julia runtests.jl failedtest --seed=failseed`.

Before running this command, if one then wants to re-run the particular test in another context (e.g. at the REPL while debugging), one can insert in failedtest.jl right before the failing test the following line to dump the RNG state: `write(open("/tmp/RNG_RESTORE.jl", "w"), "copy!(Base.GLOBAL_RNG, $(Base.GLOBAL_RNG))")`. Then to restore the state (e.g. at the REPL): `include("/tmp/RNG_RESTORE.jl")`. 
This almost implements https://github.com/JuliaLang/julia/pull/8339#issuecomment-55588256, except that the state is not printed automatically.
This is not very clean, so I hesitate to document it (and if so, where?).
Note: this depends on ~the unmerged~  #16919 for `copy!`.

I am not familiar at all with the file runtests.jl, so careful review is needed.

### CODE REVIEW

File: test/runtests.jl [hunk 1]
```diff
@@ -53,8 +53,8 @@ cd(dirname(@__FILE__)) do
                             p = addprocs(1; exeflags=`--check-bounds=yes --depwarn=error`)[1]
                             remotecall_fetch(()->include("testdefs.jl"), p)
                         else
-                            # single process testing, bail if mem limit reached, or, on an exception.
-                            isa(resp, Exception) ? rethrow(resp) : error("Halting tests. Memory limit reached : $resp > $max_worker_rss")
+                            # single process testing, bail if mem limit reached.
+                            isa(resp, Exception) || error("Halting tests. Memory limit reached : $resp > $max_worker_rss")
```
Comment (tkelman) [inline test/runtests.jl]: what's the reason for this change?

File: test/runtests.jl [hunk 2]
```diff
@@ -76,8 +77,9 @@ cd(dirname(@__FILE__)) do
     n > 1 && rmprocs(workers(), waitfor=5.0)
 
     for t in node1_tests
-        n > 1 && print("\tFrom worker 1:\t")
-        runtests(t)
+        n > 1 && print("\tFrom worker 1 :\t")
+        println("(The global RNG seed is 0x$(hex(seed)))")
```
Comment (tkelman) [inline test/runtests.jl]: this is a bit noisy to print even on success

File: test/runtests.jl (hunk 1 above)
Comment (rfourquet) [inline test/runtests.jl]: To limit the number of places in the code where the printing of the seed has to be handled. As far as I could understand, the logic is not really modified here.

File: test/runtests.jl (hunk 2 above)
Comment (rfourquet) [inline test/runtests.jl]: You are right, will change.

**Comment by rfourquet** [comment]
With reference to that [comment](https://github.com/JuliaLang/julia/pull/16940#issuecomment-226146805):
> we can run systematically the whole testsuite by providing a fixed seed (or a set thereof) on the command line, known to pass

Would it make sense here to allow multiple "--seed=SEED" command line args, which would run the tests once for each provided seed? (and for example "--seed" without an explicit seed would run with a random seed, like when there is no "--seed" option passed).

**Comment by rfourquet** [comment]
Rebased: I will merge in a few days if no objection.
